lib/src/Uncertainty/Algorithm/Sensitivity/HSICEstimatorImplementation.cxx

```diff
@@ -0,0 +1,419 @@
+
+//                                               -*- C++ -*-
+/**
+ * @brief HSICEstimatorImplementation implements the HSIC sensivity indices.
+ *
+ *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/HSICEstimatorImplementation.hxx"
+#include "openturns/HSICEstimator.hxx"
+#include "openturns/Pointer.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/Exception.hxx"
+#include "openturns/Log.hxx"
+#include "openturns/HSICStat.hxx"
+
+#include "openturns/Cloud.hxx"
+#include "openturns/Curve.hxx"
+#include "openturns/Pie.hxx"
+#include "openturns/Text.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+CLASSNAMEINIT(HSICEstimatorImplementation)
+
+static const Factory<HSICEstimatorImplementation> Factory_HSICEstimatorImplementation;
+
+typedef Collection<CovarianceModel>   CovarianceModelCollection;
+
+/** Default constructor */
+HSICEstimatorImplementation::HSICEstimatorImplementation(): PersistentObject(),
+  covarianceList_(CovarianceModelCollection()),
+  inputSample_(Sample()),
+  outputSample_(Sample()),
+  estimatorType_(HSICStat()),
+  weightFunction_(Function()),
+  n_(0),
+  inputDimension_(0),
+  HSIC_XY_(Point()),
+  HSIC_XX_(Point()),
+  HSIC_YY_(Point()),
+  R2HSICIndices_(Point()),
+  PValuesPermutation_(Point()),
+  permutationSize_(0)
+{
+  // Nothing
+}
+
+
+/** Default constructor */
+HSICEstimatorImplementation::HSICEstimatorImplementation(const CovarianceModelCollection & covarianceList, const Sample & X, const Sample & Y,
+    const HSICStat & estimatorType, const Function & weightFunction ): PersistentObject(),
+  covarianceList_(covarianceList),
+  inputSample_(X),
+  outputSample_(Y),
+  estimatorType_(estimatorType),
+  weightFunction_(weightFunction),
+  n_(X.getSize()),
+  inputDimension_(X.getDimension()),
+  HSIC_XY_(Point()),
```

Comment:
HSIC_XY_ is already a Point, so HSIC_XY_() would be enough, or it can completely omitted